docs/proposals/multi-arch-image-mgmt.md

```diff
@@ -0,0 +1,239 @@
+# Create an index from source manifests tagged 'linux-amd64' and 'linux-arm64', and push without tagging:
+oras manifest index create localhost:5000/hello linux-amd64 linux-arm64
+
+# Create an index from source manifests tagged 'linux-amd64' and 'linux-arm64', and push with the tag 'v1':
+oras manifest index create localhost:5000/hello:v1 linux-amd64 linux-arm64
+
+# Create an index from source manifests using both tags and digests, and push with tag 'v1':
+oras manifest index create localhost:5000/hello:v1 linux-amd64 sha256:99e4703fbf30916f549cd6bfa9cdbab614b5392fbe64fdee971359a77073cdf9
+
+# Create an index and push it with multiple tags:
+oras manifest index create localhost:5000/hello:tag1,tag2,tag3 linux-amd64 linux-arm64 sha256:99e4703fbf30916f549cd6bfa9cdbab614b5392fbe64fdee971359a77073cdf9
+
+# Create and push an index with annotations:
+oras manifest index create localhost:5000/hello:v1 linux-amd64 --annotation "key=val"
+
+# Create an index and push to an OCI image layout folder 'layout-dir' and tag with 'v1':
+oras manifest index create layout-dir:v1 linux-amd64 sha256:99e4703fbf30916f549cd6bfa9cdbab614b5392fbe64fdee971359a77073cdf9
+
+# Create an index and save it locally to index.json, auto push will be disabled:
+oras manifest index create --output index.json localhost:5000/hello linux-amd64 linux-arm64
+
+# Create an index and output the index to stdout, auto push will be disabled:
+oras manifest index create localhost:5000/hello linux-arm64 --output - --pretty
+
+```
+
+### Update a multi-arch image
+
+```bash
+# Remove a manifest and add two manifests from an index tagged 'v1'. The tag will point to the updated index:
+oras manifest index update localhost:5000/hello:v1 --add linux-amd64 --add linux-arm64 --remove sha256:99e4703fbf30916f549cd6bfa9cdbab614b5392fbe64fdee971359a77073cdf9
+
+# Create a new index by updating an existing index specified by its digest:
+oras manifest index update localhost:5000/hello@sha256:99e4703fbf30916f549cd6bfa9cdbab614b5392fbe64fdee971359a77073cdf9 --add linux-amd64 --remove sha256:fd6ed2f36b5465244d5dc86cb4e7df0ab8a9d24adc57825099f522fe009a22bb
+
+# Merge manifests from the index 'v2-windows' to the index 'v2':
+oras manifest index update localhost:5000/hello:v2 --merge v2-windows
+
+# Update an index and tag the updated index as 'v2.1.0' and 'v2':
+oras manifest index update localhost:5000/hello@sha256:99e4703fbf30916f549cd6bfa9cdbab614b5392fbe64fdee971359a77073cdf9 --add linux-amd64 --tag "v2.1.0" --tag "v2"
+
+# Update an index and save it locally to index.json, auto push will be disabled:
+oras manifest index update --output index.json localhost:5000/hello:v2 --add v2-linux-amd64
+
+# Update an index and output the index to stdout, auto push will be disabled:
+oras manifest index update --output - --pretty localhost:5000/hello:v2 --remove sha256:99e4703fbf30916f549cd6bfa9cdbab614b5392fbe64fdee971359a77073cdf9
+```
+
+### Inspect a multi-arch image
+
+Add two alias `show` and `inspect`:
```

Comment:
Can we elaborate more on why this is needed? Is it to inform users the purpose of the command or is it more intuitive for users to input the command?

Reply:
Yes. The purpose is to make the inspection operation more intuitive to users. `fetch` or `get` is more like download operation to users.